Assets/Scripts/Models/Furniture.cs

```diff
@@ -342,7 +342,7 @@ public static Furniture PlaceInstance(Furniture proto, Tile tile)
 
     public void Update(float deltaTime)
     {
-        if (HasPower() == false && PowerConnection.IsPowerConsumer)
+        if (PowerConnection.IsPowerConsumer && HasPower() == false)
```

Comment:
This doesn't work as well as expected because PowerConnection can be null. I very quickly get 150 warnings. You can add a check for `PowerConnection != null` or change it back.